ports/stm32/boards/ARDUINO_GIGA/stm32h747.ld

```diff
@@ -14,10 +14,10 @@ MEMORY
     SRAM4 (xrw)     : ORIGIN = 0x38000000, LENGTH = 64K     /* SRAM4 D3    */
     FLASH (rx)      : ORIGIN = 0x08000000, LENGTH = 2048K   /* Total available flash */
     FLASH_EXT (rx)  : ORIGIN = 0x90000000, LENGTH = 16384K  /* 16MBs external QSPI flash */
-    FLASH_FS (r)    : ORIGIN = 0x08020000, LENGTH = 128K    /* sector  1 -> Flash storage */
-    FLASH_TEXT (rx) : ORIGIN = 0x08040000, LENGTH = 1792K   /* Sector  0 -> Arduino Bootloader
-                                                               Sector  1 -> Reserved for CM4/FS
-                                                               Sectors 2 -> 15 firmware */
+    FLASH_FS (r)    : ORIGIN = 0x08020000, LENGTH = 128K    /* 0x08000000 -> 0x08020000 Arduino Bootloader */
+    FLASH_TEXT (rx) : ORIGIN = 0x08040000, LENGTH = 1280K   /* 0x08020000 -> 0x08040000 Flash filesystem storage */
+                                                            /* 0x08040000 -> 0x08180000 CM7 firmware */
+                                                            /* 0x08180000 -> 0x08200000 CM4 firmware */
```

Comment:
I'm not sure about this flash layout, it seems that there is some wasted space for `FLASH_FS` which is unused on this board.  Or is it required by the bootloader than the application starts at `0x08040000`?

Either way, it would be good to make this clearer with named sections, eg:
```
FLASH_BL (rx)    : ORIGIN = 0x08000000, LENGTH = 128K /* Arduino bootloader */
FLASH_UNUSED (r) : ORIGIN = 0x08020000, LENGTH = 128K /* unused? */
FLASH_APP (rx)   : ORIGIN = 0x08040000, LENGTH = 1280K /* CM7 firmware */
FLASH_CM4 (rx)   : ORIGIN = 0x08180000, LENGTH = 512K /* CM4 firmware */
```

Then use `common_bl.ld` instead of `common_blifs.ld`.

And can get rid of `_micropy_hw_internal_flash_storage_ram_cache_start` and `_micropy_hw_internal_flash_storage_start`.

Is the address 0x08180000 completely fixed, eg other non-MicroPython Arduino firmware also uses this segmentation?  I'm just wondering if 512K is a good value for the size of the CM4 firmware.  It looks reasonable to me, but once this is decided then it'll be hard to change without making it hard for users.

(Same comments go for the other 2 Arduino STM boards.)

Reply:
The bootloader on all of those boards jumps to `0x08040000`, so we can't use that sector for code space, but we're free to use it for anything else. Currently it's only good for flash FS, when the external storage is disabled, but in the future it can be repurposed for romfs for example.
I don't think there's any good reason to remove the internal FS config right now, but I don't mind it, but I would rather revisit this in a follow-up PR, because it's not really related to this one.

> Is the address 0x08180000 completely fixed, eg other non-MicroPython Arduino firmware also uses this segmentation?

Yes the Arduino firmware uses this address when the flash is configured for a 75-25 split. There's a 50-50 split that won't be supported. Note that this address is not actually used in the firmware or hard-coded, it's more of a guideline to reserve some minimum flash space for CM4.

Reply:
> I don't think there's any good reason to remove the internal FS config right now, but I don't mind it, but I would rather revisit this in a follow-up PR, because it's not really related to this one.

OK, that's fine.

But in that case can you please update these ld files as follows (because at the moment the comments don't match the line they are describing):
```
FLASH_BL (rx)    : ORIGIN = 0x08000000, LENGTH = 128K /* Arduino bootloader */
FLASH_FS (r)     : ORIGIN = 0x08020000, LENGTH = 128K /* filesystem */
FLASH_TEXT (rx)  : ORIGIN = 0x08040000, LENGTH = 1280K /* CM7 firmware */
FLASH_CM4 (rx)   : ORIGIN = 0x08180000, LENGTH = 512K /* CM4 firmware */
```